Add changes for edge-22.12.1 (#9926)
linkerd/linkerd2: CHANGES.md

# Changes

## edge-22.12.1

This edge release introduces static and dynamic port overrides for CNI eBPF
socket-level load balancing. In certain installations when CNI plugins run in
eBPF mode, socket-level load balancing rewrites packet destinations to port
6443; as with 443 already, this port is now skipped as well on control plane
components so that they can communicate with the Kubernetes API before their
proxies are running.

Additionally, a potential panic and false warning have been fixed in the

> destination component.

Reviewer: ```suggestion
destination controller.
```

> * Updated linkerd-jaeger's collector component to expose port 4318 in order

Reviewer: ```suggestion
* Updated linkerd-jaeger's collector to expose port 4318 in order
```

  support HTTP alongside gRPC (thanks @uralsemih!)

> * Introduced the `privileged` configuration which allows the `proxy-init`
>   container to run as privileged without also running as root

Reviewer: > Introduced the `privileged` configuration

to what?

```suggestion
* Added a `proxyInit.privileged` setting to control whether the
  init container runs as a privileged process
```

> * Fixed a potential panic in the destination component caused by concurrent

Reviewer: ```suggestion
* Fixed a potential panic in the destination controller caused by concurrent
```
